### ISSUE TITLE
I can't import a machine with a clone block in it

### ISSUE THREAD

**Issue by djhaskin987** [terraform-providers/terraform-provider-vsphere#333]
### Terraform Version

```
Terraform v0.11.1
```

### vSphere Provider Version

```
.
├── provider.vsphere 1.1.1
├── provider.zenoss
└── module.ranchermgr
    ├── provider.null
    └── module.vsphere_instance
        ├── provider.external
        ├── provider.template
        ├── provider.vsphere (inherited)
        └── provider.zenoss (inherited)
```

### Affected Resource(s)

- `vsphere_virtual_machine`

### Configuration File and Steps to Reproduce

Basically, I have a `vsphere_virtual_machine` block that has a `clone` block in it.

I do a `terraform state rm <virtual-machine-object>` and reimport it. 
Finally, I run a `terraform plan`. I get an error message generated by [this line of code](https://github.com/terraform-providers/terraform-provider-vsphere/blob/c14f9ace38c6a0d5ab5599064a32f1b8e933c0d7/vsphere/resource_vsphere_virtual_machine.go#L539).

### Expected Behavior

It should say "nothing to do" on the terraform plan

### Actual Behavior

It errors, saying I can't have a clone block

### Comments

The error message says I must either delete the `clone` block from config or not import machines. But if I delete the clone block, I can't make new ones if I need them.

I can't use terraform to import machines and then manage them afterwards.

**Comment by almoore** [comment]
+1

**Comment by vancluever** [comment]
@djhaskin987 and @almoore, this is currently intended behaviour. Importing does not populate the `clone` sub-resource at all, hence on the next plan, it would force a new resource. This is blocked explicitly to prevent accidental deletion.

There have been some offline requests to change this behaviour and I can see the value in not duplicating your configs, as such I will mark this as an enhancement to see if we can find a better way. The challenge lies in the complexity of the structure of `clone` and `customize` and what we can do to ensure that block is ignored in the diff, yet still allow changes to force a new resource naturally.

**Comment by djhaskin987** [comment]
This means that either I can import *all* my VMs, or I can clone them, but I can't do both. I can't tell Terraform how to make a new VM and also tell Terraform that a previously created one exists. 

I understand that this is a complicated task and the vsphere soap api doesn't make it easier, but I can't really use this provider (or pitch it to the ops team) if I don't have the assurance that I can import resources while also being able to recreate them.

**Comment by vancluever** [comment]
@djhaskin987 to be fair, this is more of a TF issue than it is a vSphere issue. :slightly_smiling_face: 

I have some ideas around how this can be implemented, so it might be reasonable to see this in 1.2.0 or 1.3.0 in the new year. Stay tuned!

**Comment by djhaskin987** [comment]
"not duplicating your configs" - does that mean there's a workaround involving using a few extra config blocks? If so, I welcome it and would happily use it instead. Is there such a workaround?

**Comment by vancluever** [comment]
@djhaskin987 yeah, importing those VMs with a config that doesn't have the `clone` block in it will be okay.

What I would do (personally, not saying this applies to your org) is just run the current projects out of a different config repo for now, maybe using a variation of your `ranchermgr` module that has had the `clone` block removed, but with the rest of the config intact.

Even further, if you are managing all of this with a module, it might just be as easy as forking the module for now, switching to that one, and then switching back either when you need to re-deploy those VMs (at that point, you should be able to taint the resources with `terraform taint`), or when we remove the restriction in a later release.

So sorta more config, and sorta less config. :slightly_smiling_face: But if you can deal with the divergence for now, it should work for you.

**Comment by vancluever** [comment]
Hey all,

We just released 1.4.0 which lifts the import and legacy upgrade restrictions on the `clone` sub-resource block. There a few things you should know about how this functionality works (namely pertaining to configuration changes on the first apply post-import), so head over to [Additional requirements and notes for
  importing](https://www.terraform.io/docs/providers/vsphere/r/virtual_machine.html#additional-requirements-and-notes-for-importing) in the resource documentation for more details on that. Otherwise, I hope this new functionality is useful and let us know if anyone has any issues!
